docs/dev.md

```diff
@@ -64,6 +71,67 @@ dep ensure
 /go/bin/deepcopy-gen --output-base=/go/src --input-dirs=github.com/kubeflow/experimental-kvc/pkg/apis/cr/v1/...
 go build -gcflags "-N -l" github.com/kubeflow/experimental-kvc
 ```
+### Developing on a workstation
+Prior to using this method please ensure you have a `go 1.9` or better development environment setup and running.
```

Comment:
Why would we need this apart from just docker and docker_make?

Reply:
@Ajay191191 The reason is very simple: Why do I need the overhead introduced by `docker` when I'm already sitting on a `workstation` that I own:

#### Running on docker
```
$ time ./docker_make all
go test -v --cover ./pkg/resource/...
=== RUN   TestGenericClient
--- PASS: TestGenericClient (0.00s)
	genericClient_test.go:125: Testing for pod
	genericClient_test.go:125: Testing for node
	genericClient_test.go:125: Testing for pv
	genericClient_test.go:125: Testing for pvc
PASS
coverage: 64.1% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/resource	0.047s	coverage: 64.1% of statements
=== RUN   TestReify
--- PASS: TestReify (0.00s)
	reify_test.go:90: TestReify: empty template, no values
	reify_test.go:90: TestReify: non-empty template, no values
	reify_test.go:90: TestReify: expansion from values object
	reify_test.go:90: TestReify: expansion of resource string from values object
	reify_test.go:90: TestReify: invalid template syntax
	reify_test.go:90: TestReify: invalid template value reference
PASS
coverage: 100.0% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/resource/reify	0.068s	coverage: 100.0% of statements
go test -v --cover ./pkg/hooks/...
=== RUN   TestHook
--- PASS: TestHook (0.00s)
PASS
coverage: 44.3% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/hooks	0.038s
go test -v --cover ./pkg/handlers/...
=== RUN   TestHandler
--- PASS: TestHandler (0.00s)
	handler_test.go:254: Testing for: [s3_dev_handler] Any create failed
	handler_test.go:254: Testing for: [s3_handler] labels not set
	handler_test.go:254: Testing for: [s3_handler] replicas > Num nodes
	handler_test.go:254: Testing for: [nfs_handler] Any create failed
	handler_test.go:254: Testing for: [s3_handler] Wrong timeoutForDataDownload format
	handler_test.go:254: Testing for: [nfs_handler] server not set
	handler_test.go:254: Testing for: [s3_dev_handler] labels not set
	handler_test.go:254: Testing for: [s3_dev_handler] Wrong timeoutForDataDownload format
	handler_test.go:254: Testing for: [s3_dev_handler] replicas > Num nodes
	handler_test.go:254: Testing for: [s3_handler] Wrong access mode
	handler_test.go:254: Testing for: [nfs_handler] labels not set
	handler_test.go:254: Testing for: [nfs_handler] path not set
	handler_test.go:254: Testing for: [nfs_handler] Wrong access mode
	handler_test.go:254: Testing for: [s3_dev_handler] awsCredentialsSecretName not set
	handler_test.go:254: Testing for: [s3_dev_handler] Wrong access mode
	handler_test.go:254: Testing for: [s3_dev_handler] Node List Failing
	handler_test.go:254: Testing for: [s3_handler] Any create failed
	handler_test.go:254: Testing for: [s3_handler] awsCredentialsSecretName not set
	handler_test.go:254: Testing for: [s3_handler] Node List Failing
PASS
coverage: 41.7% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/handlers	0.043s
go test -v --cover ./pkg/controller/...
=== RUN   TestController
Started watching for VolumeManager CR objects.
--- PASS: TestController (0.00s)
PASS
coverage: 100.0% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/controller	0.047s

real	2m10.131s
user	0m0.182s
sys	0m0.077s
```
#### Running on Workstaion
```
$ time make all
go test -v --cover ./pkg/resource/...
=== RUN   TestGenericClient
--- PASS: TestGenericClient (0.02s)
	genericClient_test.go:125: Testing for pod
	genericClient_test.go:125: Testing for node
	genericClient_test.go:125: Testing for pv
	genericClient_test.go:125: Testing for pvc
PASS
coverage: 64.1% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/resource	(cached)	coverage: 64.1% of statements
=== RUN   TestReify
--- PASS: TestReify (0.00s)
	reify_test.go:90: TestReify: empty template, no values
	reify_test.go:90: TestReify: non-empty template, no values
	reify_test.go:90: TestReify: expansion from values object
	reify_test.go:90: TestReify: expansion of resource string from values object
	reify_test.go:90: TestReify: invalid template syntax
	reify_test.go:90: TestReify: invalid template value reference
PASS
coverage: 100.0% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/resource/reify	(cached)	coverage: 100.0% of statements
go test -v --cover ./pkg/hooks/...
=== RUN   TestHook
--- PASS: TestHook (0.00s)
PASS
coverage: 44.3% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/hooks	(cached)	coverage: 44.3% of statements
go test -v --cover ./pkg/handlers/...
=== RUN   TestHandler
--- PASS: TestHandler (0.00s)
	handler_test.go:254: Testing for: [nfs_handler] labels not set
	handler_test.go:254: Testing for: [nfs_handler] server not set
	handler_test.go:254: Testing for: [s3_dev_handler] labels not set
	handler_test.go:254: Testing for: [s3_dev_handler] Wrong access mode
	handler_test.go:254: Testing for: [s3_dev_handler] replicas > Num nodes
	handler_test.go:254: Testing for: [s3_handler] awsCredentialsSecretName not set
	handler_test.go:254: Testing for: [s3_handler] Wrong timeoutForDataDownload format
	handler_test.go:254: Testing for: [s3_handler] Node List Failing
	handler_test.go:254: Testing for: [nfs_handler] Wrong access mode
	handler_test.go:254: Testing for: [s3_handler] Wrong access mode
	handler_test.go:254: Testing for: [s3_dev_handler] Wrong timeoutForDataDownload format
	handler_test.go:254: Testing for: [s3_dev_handler] Node List Failing
	handler_test.go:254: Testing for: [s3_dev_handler] Any create failed
	handler_test.go:254: Testing for: [s3_handler] labels not set
	handler_test.go:254: Testing for: [s3_handler] replicas > Num nodes
	handler_test.go:254: Testing for: [s3_dev_handler] awsCredentialsSecretName not set
	handler_test.go:254: Testing for: [s3_handler] Any create failed
	handler_test.go:254: Testing for: [nfs_handler] path not set
	handler_test.go:254: Testing for: [nfs_handler] Any create failed
PASS
coverage: 41.7% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/handlers	(cached)	coverage: 41.7% of statements
go test -v --cover ./pkg/controller/...
=== RUN   TestController
Started watching for VolumeManager CR objects.
--- PASS: TestController (0.00s)
PASS
coverage: 100.0% of statements
ok  	github.com/kubeflow/experimental-kvc/pkg/controller	(cached)	coverage: 100.0% of statements

real	0m2.625s
user	0m2.250s
sys	0m1.945s 
```

I found it hard to argue with those numbers 🙂

Reply:
The timing is not an issue here, but if we introduce 2 methods, it's not clear looking at the docs, when should I use the docker method vs when should I use the other one. Maybe we can make that clearer? For example, if you have docker installed and don't want to install go and it's dependent packages and also handle the go env vars, use the first method, but if for some reason you don't have access to docker or can't do docker, use the second method. Also why does a github have to be setup?

Reply:
Fair enough, let me update it @Ajay191191